### DISCUSSION TITLE
Raw queries work somewhat unreasonable

### DISCUSSION THREAD

**Discussion by kasir-barati** [prisma/prisma#25629] [Q&A] [answered]
### Question

So I have a pretty simple query:

```sql
SELECT id, created_at as "createdAt", updated_at as "updatedAt", name, email
FROM public.users
WHERE name ILIKE '%121%';
```

I can run this query in my database and get the data I wanted. It has no SQL syntactical issue. So with that out of the way I have written the very exact query in Prisma like this: 

https://github.com/kasir-barati/react/blob/1e5b46247738dc1d1f20fe5660cb6fda443fdec1/apps/api/src/repositories/user.repository.ts#L15-L21

Just copy and paste it with the query that is running right now. And you will receive nothing. But you can open your terminal and see that it is actually passing the correct parameter to the query. So what am I missing here?

### How to reproduce (optional)

1. Clone https://github.com/kasir-barati/react/
2. Run `pnpm i`.
3. Change the query which is being executed in `user.repository.ts` inside the `getUsers` to the one that is commented.
4. Run `nx prisma-push api`.
5. Run `nx prisma-seed api`.
6. Run `nx serve api`.
7. Run `nx serve react`.
8. Open localhost:4200 and type in the search box. Top, left corner.

See the logs inside the terminal.

### Expected behavior (optional)

Just return the matched records. That's why I used `ILIKE %name%` query.

Note that exact match works but there is something going on with my ILIKE query.

**Comment by aqrln** [comment]
It looks like you have extra quotes around `${whereCondition}` in your query

**Reply by kasir-barati** [reply]
Without that extra quote SQL will yell at me saying that there is a syntax issue in my SQL query. So the final result would be something like this:

```sql
SELECT id, created_at as "createdAt", updated_at as "updatedAt", name, email
FROM public.users
WHERE name ILIKE '%something%';
```

But without quotation mark PSQL is gonna be upset with my query:

```sql
SELECT id, created_at as "createdAt", updated_at as "updatedAt", name, email
FROM public.users
WHERE name ILIKE %something%;
```

There is nothing wrong with the query with quotation mark. The issue is that when I just opy and paste same query in pgadmin I get the expected result while Prisma (or IDK someone in between) is making a mistake.

**Reply by ludralph** [reply, marked as answer]
Hi @kasir-barati 

Can you modify your code to look like the one below?
```js
const whereCondition = 'John'; // replace 'John' with your variable
const result = await prisma.$queryRaw`
      SELECT id, created_at as "createdAt", updated_at as "updatedAt", name, email
      FROM public.users
      WHERE name ILIKE ${`%${whereCondition}%`}`;
```
You can also use Prisma built-in [filtering](https://www.prisma.io/docs/orm/reference/prisma-client-reference#mode)
```js
const result = await prisma.user.findMany({
  where: {
    name: {
      contains: name,
      mode: 'insensitive'
    }
  },
  select: {
    id: true,
    createdAt: true,
    updatedAt: true,
    name: true,
    email: true
  }
});
```

**Reply by kasir-barati** [reply]
Thanks @ludralph, I am curious as what was happening with my query and why your solution works. I do not remember accurately but I guess I also tried `'%${name}%' too but it did not work. Maybe I am missing something or maybe it was due to another but that was in my seed script.

**Reply by ludralph** [reply]
Your query directly concatenates the `whereCondition` variable into the SQL string. The one I shared uses a tagged template literal to dynamically construct the SQL query. [queryRaw](https://www.prisma.io/docs/orm/prisma-client/using-raw-sql/raw-queries#queryraw) in Prisma will only accept a template string.
